typing/typecore.ml

```diff
@@ -5716,7 +5717,7 @@ let report_error ~loc env = function
         if b then
           fprintf ppf ".@.@[<hov>%s@ %s@ %s@]"
             "This simple coercion was not fully general."
-            "Hint: Consider using a fully explicit coercion"
+            "@{<hint>Hint@}: Consider using a fully explicit coercion"
```

Comment:
This hint needs to be inlined inside the format string, otherwise it will not be formatted.